falcon/media/json.py

```diff
@@ -12,19 +12,88 @@ class JSONHandler(BaseHandler):
 
     This handler uses Python's :py:mod:`json` by default, but will
     use :py:mod:`ujson` if available.
+
+    Keyword Arguments:
+        dumps (func): the function to use when serializing JSON responses
+
+        loads (func): the function to use when deserializing JSON requests
+
+        dumps_kwargs (dict): keyword arguments passed to ``dumps`` when
+            serializing to JSON.  ``ensure_ascii`` is set to ``false`` by
+            default.  To override this setting, pass ``{ensure_ascii=True}``
+            explicitly. You can completely disable extra parameters by passing
+            ``False``.
+
+        loads_kwargs (func): keyword arguments to pass to ``loads`` when
+            deserializing from JSON.  You can completely disable extra
+            parameters by passing ``False``.
+
+    You can override the JSON library used by changing the ``dumps`` and
+    ``loads`` functions::
+
+        import falcon
+        from falcon import media
+        import rapidjson
+
+
+        json_handler = media.JSONHandler(
+            dumps=rapidjson.dumps,
+            loads=rapidjson.loads,
+            dumps_kwargs=dict(ensure_ascii=False)
+            loads_kwargs=False,
+        )
+        extra_handlers = {
+            'application/json': json_handler,
+            'application/json; charset=UTF-8': json_handler,
+        }
+
+        api = falcon.API()
+        api.req_options.media_handlers.update(extra_handlers)
+        api.resp_options.media_handlers.update(extra_handlers)
     """
 
+    dumps_kwargs = False
+    loads_kwargs = False
+
+    def __init__(self, dumps=None, dumps_kwargs=None, loads=None, loads_kwargs=None):
+        self.dumps = dumps or json.dumps
+
+        # NOTE(nZac): Turn off kwargs support by manually passing `False` to
+        # either `_kwargs` argument.
+        if dumps_kwargs is not False:
```

Comment:
This works, but to be honest I would prefer another solution than giving ``False`` a special meaning this way, provided a cleaner solution exists, of course.

As an idea, how about having empty *kwargs* as defaults, and instead configuring the desired ones inside ``class Handlers(UserDict)`` constructor (``falcon/media/handlers.py``)?